calico_rkt/calico_rkt.py

```diff
@@ -1,198 +1,212 @@
+                                               interface=interface,
+                                               subnet=subnet)
 
-        _create_profile(endpoint=endpoint, 
-                        profile_name=input_json['name'], 
-                        ip=ip,
-                        client=_datastore_client)
+        _set_profile_on_endpoint(endpoint=endpoint,
+                                 profile_name=net_name,
+                                 ip=ip)
     except CalledProcessError as e:
-        print_stderr('Error code %d creating pod networking: %s\n%s' % (
+        _log.error('Error code %d creating pod networking: %s\n%s' % (
             e.returncode, e.output, e))
         sys.exit(1)
-    print_stderr('Finished Creating pod %s' % container_id)    
 
-def delete(container_id):
+    _log.info('Finished Creating pod %s' % container_id)
+
+
+def delete(args):
     """Cleanup after a pod."""
-    print_stderr('Deleting pod %s' % container_id, file=sys.stderr)
+    container_id = args['container_id']
+    net_name = args['name']
 
-    _datastore_client = IPAMClient()
+    _log.info('Deleting pod %s' % container_id)
 
     # Remove the profile for the workload.
     _container_remove(hostname=HOSTNAME,
-                        orchestrator_id=ORCHESTRATOR_ID,
-                        container_id=container_id,
-                        client=_datastore_client)
-
-    # Delete profile
-    try:
-        _datastore_client.remove_profile(input_json['name'])
-    except:
-        print_stderr("Cannot remove profile %s; Profile cannot be found." % container_id)
-
-def _create_calico_endpoint(container_id, netns_path, client):
-    """Configure the Calico interface for a pod."""
-    print_stderr('Configuring Calico networking.', file=sys.stderr)
-
-    interface = env['CNI_IFNAME']
-
-    endpoint, ip = _container_add(hostname=HOSTNAME,
-                                orchestrator_id=ORCHESTRATOR_ID,
-                                container_id=container_id,
-                                netns_path=netns_path,
-                                interface=interface,
-                                client=client)
-    print_stderr('Finished configuring network interface', file=sys.stderr)
-    return endpoint, ip
-
-def _container_add(hostname, orchestrator_id, container_id, netns_path, interface, client):
+                      orchestrator_id=ORCHESTRATOR_ID,
+                      container_id=container_id)
+
+    # Delete profile if only member
+    if _datastore_client.profile_exists(net_name) and \
+       len(_datastore_client.get_profile_members(net_name)) < 1:
+        try:
+            _log.info(
+                "Profile %s has no members, removing from datastore" % net_name)
+            _datastore_client.remove_profile(net_name)
+        except:
+            _log.error(
+                "Cannot remove profile %s: Profile cannot be found." % container_id)
```

Comment:
I think we have previously agreed that the pep8 limit should be 99 chars
for Calico projects. But if you hit that limit in an error message, just do
a line break somewhere in the string.

There are a couple ways to do that;
```
_log.error("abc"
              "def")
```
works, because of Python's string concatenation rules.

Or,

```
_log.error("abc %s" \
               % var)
```
Should also work. It's OK to break pep8 in test code where it aids clarity, but we're generally not OK with breaking it in normal code.